packages/providers/mgt-msal2-provider/src/Msal2Provider.ts

```diff
@@ -0,0 +1,463 @@
+      this._domainHint = typeof config.domain_hint !== 'undefined' ? config.domain_hint : null;
+      this.scopes = typeof config.scopes !== 'undefined' ? config.scopes : ['user.read'];
+      this._publicClientApplication = new PublicClientApplication(this.ms_config);
+      this.multiAccountDisabled =
+        typeof config.multiAccountDisabled !== 'undefined' ? config.multiAccountDisabled : true; //Set this to true for now. Once multi account is enabled, default will be false
+      try {
+        this._publicClientApplication.handleRedirectPromise().then((tokenResponse: AuthenticationResult | null) => {
+          this.handleResponse(tokenResponse);
+          if (tokenResponse === null) {
+            this.trySilentSignIn();
+          }
+        });
+      } catch (e) {
+        throw e;
+      }
+    } else {
+      throw new Error('clientId must be provided');
+    }
+
+    this.graph = createFromProvider(this);
+  }
+
+  /**
+   * Attempts to sign in user silently
+   *
+   * @memberof Msal2Provider
+   */
+  public async trySilentSignIn() {
+    let silentRequest: any = {
+      scopes: this.scopes,
+      domainHint: this._domainHint
+    };
+    if (this._sid || this._loginHint) {
+      silentRequest.sid = this._sid;
+      silentRequest.loginHint = this._loginHint;
+    } else {
+      this._account = this.getAccount();
+      if (this._account) {
+        silentRequest.sid = this._account.idTokenClaims.sid || null;
+        silentRequest.loginHint = this._account.idTokenClaims.preferred_username;
+      }
+    }
+    if (silentRequest.sid || silentRequest.loginHint) {
+      try {
+        this.setState(ProviderState.Loading);
+        const response = await this._publicClientApplication.ssoSilent(silentRequest);
+        if (response) {
+          this.handleResponse(response);
+        }
+      } catch (e) {
+        this.setState(ProviderState.SignedOut);
+      }
+    }
+  }
+
+  /**
+   * Log in the user
+   *
+   * @return {*}  {Promise<void>}
+   * @memberof Msal2Provider
+   */
+  public async login(): Promise<void> {
+    const loginRequest: PopupRequest = {
+      scopes: this.scopes,
+      loginHint: this._loginHint,
+      prompt: 'select_account',
+      domainHint: this._domainHint
+    };
+    if (this._loginType == LoginType.Popup) {
+      const response = await this._publicClientApplication.loginPopup(loginRequest);
+      this.handleResponse(response);
+    } else {
+      const loginRedirectRequest: RedirectRequest = { ...loginRequest };
+      this._publicClientApplication.loginRedirect(loginRedirectRequest);
+    }
+  }
+
+  /**
+   * Get all signed in accounts
+   *
+   * @return {*}
+   * @memberof Msal2Provider
+   */
+  public getAllAccounts() {
+    let usernames = [];
+    this._publicClientApplication.getAllAccounts().forEach((account: AccountInfo) => {
+      usernames.push({ username: account.username, id: account.homeAccountId });
```

Comment:
This (`{username: string, id: string}`) can be the type for the IProvider for these methods instead of `any`

Reply:
This is meant to be sent to mgt-login for rendering the logged in user details, thought I'd wait till the design is done so I know what details have to be sent

Reply:
I'm ok with that if you are already planning to update later